pkg/dev/admin.go

```diff
@@ -94,16 +93,27 @@ func (m *adminManager) clusterRoleBinding(clusterRoleBinding *rbacv1.ClusterRole
 	clusterRoleBinding.Subjects = []rbacv1.Subject{
 		{
 			Kind:      "ServiceAccount",
-			Name:      relayAdminServiceAccountName,
-			Namespace: systemNamespace,
+			Name:      m.objects.serviceAccount.Name,
+			Namespace: m.objects.serviceAccount.Namespace,
 		},
 	}
 }
 
 func (m *adminManager) addConnectionForWorkflow(ctx context.Context, name string) error {
 	secretKey := client.ObjectKeyFromObject(&m.objects.secret)
 
-	if err := m.cl.APIClient.Get(ctx, secretKey, &m.objects.secret); err != nil {
+	err := retry.Wait(ctx, func(ctx context.Context) (bool, error) {
+		if err := m.cl.APIClient.Get(ctx, secretKey, &m.objects.secret); err != nil {
+			return retry.Repeat(err)
+		}
+
+		if len(m.objects.secret.Data) == 0 {
+			return retry.Repeat(nil)
```

Comment:
It might good to include an error here like `errors.New("secret data is still empty")` so you get a bit more information if the context is cancelled.

Reply:
Yeah, I went back and forth on that. I'm not a fan of creating "fake" errors in general (especially in this case when it's not technically an error), but I figured it would come up in PR... =)

I'll add one.